### PR TITLE
fix: Remove animations on page transitions

### LINKED ISSUE #546: Transition between pages (home and groups) is not looking good
**Author the user story for this feature**

As a user, I want to see smooth transitions implemented so I can have a pleasant experience using the site.

**Is your feature request related to a problem? Please describe.**
See recording:
[Screen_Recording_2020-09-09_at_12.09.01.mov.zip](https://github.com/platform-platform/monorepo/files/5224242/Screen_Recording_2020-09-09_at_12.09.01.mov.zip)

**Describe the solution you'd like**
No animations on page transitions.

**Describe alternatives you've considered**
-
**Additional context**
-

### PR DESCRIPTION
**Pull request by ghost**
Fixes #546

### CODE REVIEW

File: metrics/web/lib/common/presentation/routes/metrics_page_route.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+
+class MetricsPageRoute<T> extends MaterialPageRoute<T> {
```
Comment (vova-beloded) [inline metrics/web/lib/common/presentation/routes/metrics_page_route.dart:3]: It seems like this class lacks documentation. Also, should we add tests for this class?

File: metrics/web/lib/common/presentation/routes/metrics_page_route.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  MetricsPageRoute({
+    @required WidgetBuilder builder,
+    RouteSettings settings,
+  }) : super(
```
Comment (vova-beloded) [inline metrics/web/lib/common/presentation/routes/metrics_page_route.dart:7]: It seems like this class should accept all parameters available in the `MaterialPageRoute`. Could you please add missing parameters here?

File: metrics/web/lib/common/presentation/routes/route_generator.dart [hunk 3]
```diff
@@ -48,15 +49,15 @@ class RouteGenerator {
     );
   }
 
-  /// Creates [MaterialPageRoute] with the given [name] and the [widget].
+  /// Creates [MetricsPageRoute] with the given [name] and the [widget].
   ///
   /// Throws an [AssertionError] if the [widget] is null.
   static MaterialPageRoute _createMaterialPageRoute({
```
Comment (vova-beloded) [inline metrics/web/lib/common/presentation/routes/route_generator.dart:55]: Should we rename this method and make it return MetricsPageRoute? 
```suggestion
  static MetricsPageRoute _createMetricsPageRoute({
```

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Well done, @solid-tetianamovlian! I've added a couple of suggestions, please take a look.

File: metrics/web/test/common/presentation/routes/metrics_page_route_test.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import 'package:metrics/common/presentation/routes/metrics_page_route.dart';
+import 'package:metrics/common/presentation/routes/route_name.dart';
+import 'package:test/test.dart';
+
+import '../../../test_utils/matcher_util.dart';
+
+void main() {
+  group('MetricsPageRoute', () {
+    test(
+      'constructor throws the AssertionError if the builder is null',
```
Comment (vova-beloded) [inline metrics/web/test/common/presentation/routes/metrics_page_route_test.dart:11]: Can we remove the `constructor` from the description to be consistent with other tests in the project?
```suggestion
      'throws the AssertionError if the builder is null',
```

File: metrics/web/test/common/presentation/routes/metrics_page_route_test.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:metrics/common/presentation/routes/metrics_page_route.dart';
+import 'package:metrics/common/presentation/routes/route_name.dart';
+import 'package:test/test.dart';
+
+import '../../../test_utils/matcher_util.dart';
+
+void main() {
+  group('MetricsPageRoute', () {
```
Comment (vova-beloded) [inline metrics/web/test/common/presentation/routes/metrics_page_route_test.dart:9]: We should use the double-quoted strings for group names and test descriptions according to the [dart code style](https://github.com/platform-platform/monorepo/blob/master/docs/10_dart_code_style.md#test-descriptions) document. Also, maybe we can check that the `buildTransitions` function returns the given child to be sure that there are no animations?

File: metrics/web/test/common/presentation/routes/metrics_page_route_test.dart [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    test(
+      'constructor throws the AssertionError if the maintainState is null',
+      () {
+        expect(
+          () => MetricsPageRoute(
+              builder: (_) => Container(),
+              settings: const RouteSettings(name: RouteName.login),
+              maintainState: null),
+          MatcherUtil.throwsAssertionError,
+        );
```
Comment (vova-beloded) [inline metrics/web/test/common/presentation/routes/metrics_page_route_test.dart:42]:
```suggestion
                expect(
          () => MetricsPageRoute(
            builder: (_) => Container(),
            settings: const RouteSettings(name: RouteName.login),
            maintainState: null,
          ),
          MatcherUtil.throwsAssertionError,
        );
```

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-tetianamovlian! I've added a couple of suggestions here, let me know what do you think.

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, @solid-tetianamovlian! The code looks great now!

**Pull request merged**